toolkit/tools/pkg/imagecustomizerlib/bootcustomizer.go

```
@@ -162,29 +240,36 @@ func (b *BootCustomizer) UpdateSELinuxCommandLineForEMU(selinuxMode imagecustomi
 func (b *BootCustomizer) UpdateKernelCommandLineArgs(defaultGrubFileVarName defaultGrubFileVarName,
 	argsToRemove []string, newArgs []string,
 ) error {
-	if b.isGrubMkconfig {
+	switch b.bootConfigType {
+	case bootConfigTypeGrubMkconfig:
 		defaultGrubFileContent, err := updateDefaultGrubFileKernelCommandLineArgs(b.defaultGrubFileContent,
 			defaultGrubFileVarName, argsToRemove, newArgs)
 		if err != nil {
 			return err
 		}
 
 		b.defaultGrubFileContent = defaultGrubFileContent
-	} else {
+
+	case bootConfigTypeGrubLegacy:
 		grubCfgContent, err := updateKernelCommandLineArgsAll(b.grubCfgContent, argsToRemove, newArgs)
 		if err != nil {
 			return err
 		}
 
 		b.grubCfgContent = grubCfgContent
+
+	case bootConfigTypeUki:
+		// UKI passthrough mode: preserve existing UKI boot configuration.
```

Should we return an error here instead?